modules/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/OtelTracingContextPropagator.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import org.opensearch.telemetry.tracing.noop.NoopSpan;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BiConsumer;
+
+import static org.opensearch.telemetry.tracing.DefaultTracer.CURRENT_SPAN;
+
+/**
+ * Otel implementation of TracingContextPropagator
+ */
+public class OtelTracingContextPropagator implements TracingContextPropagator {
+
+    private static final String ROOT_SPAN = "root_span";
+
+    private final OpenTelemetry openTelemetry;
+
+    /**
+     * Creates OtelTracingContextPropagator instance
+     * @param openTelemetry Otel OpenTelemetry instance
+     */
+    public OtelTracingContextPropagator(OpenTelemetry openTelemetry) {
+        this.openTelemetry = openTelemetry;
+    }
+
+    @Override
+    public Span extractSpanFromHeader(Map<String, String> header) {
+        Context context = openTelemetry.getPropagators().getTextMapPropagator().extract(Context.current(), header, TEXT_MAP_GETTER);
+        if (context != null) {
+            io.opentelemetry.api.trace.Span span = io.opentelemetry.api.trace.Span.fromContext(context);
+            return new OTelSpan(ROOT_SPAN, span, null, Level.ROOT);
```

Comment:
Where is coming from `ROOT_SPAN` ? This is span propagated from the context, we do not have name for it, should be `null`.

Reply:
`null` name doesn't seem right. Have renamed it to `PROPAGATED_SPAN`, let me know if it works

Reply:
We probably need to have more precise abstraction like `PropagatedSpan` I think- that would be a span with no name (and me be other properties)

Reply:
Done